doc/conditional_independence.rst

```diff
@@ -0,0 +1,180 @@
+  which if we implement the naive approach using a KDTree, then it generally suffers in high-dimensions.
+  In our examples, we see it suffer with dimensionality > 4 or 5.
+
+  Estimates of CMI can be converted into a CI hypothesis test by permutation testing :footcite:`Runge2018cmi`.
+  One can generate estimated samples from the null distribution by permuting samples in an intelligent manner
+  and then the CMI value generated from the actual observed data can be compared to the CMI values computed
+  on the permutated datasets to estimate a pvalue.
+
+  It is worth noting that if one has "robust" estimates of k-NN using for example a model that
+  is effective in high-dimensions, then the KSG estimator for CMI may be quite good. For example,
+  one can use variants of Random Forests to generate adaptive nearest-neighbor estimates in high-dimensions
+  or on manifolds, such that the KSG estimator is still quite good.
+
+.. autosummary::
+   :toctree: generated/
+
+    CMITest
+
+- The Classifier Divergence approach estimates CMI using a classification model.
+
+.. autosummary::
+   :toctree: generated/
+
+    ClassifierCMITest
+
+- Direct posterior estimates can be implemented with a classification model by directly
+  estimating :math:`P(y|x)` and :math:`P(y|x,z)`, which can be used as plug-in estimates
+  to the equation for CMI.
+
+Partial (Pearson) Correlation
+-----------------------------
+Partial correlation based on the Pearson correlation is equivalent to CMI in the setting
+of normally distributed data. Computing partial correlation is fast and efficient and
+thus attractive to use. However, this **relies on the assumption of multivariate Gaussianity**,
+which may be unrealistic in certain datasets.
+
+.. autosummary::
+   :toctree: generated/
+
+    FisherZCITest
+
+Discrete, Categorical and Binary Data
+-------------------------------------
+If one has discrete data, then the test to use is based on Chi-square tests. The :math:`G^2`
+class of tests will construct a contingency table based on the number of levels across
+each discrete variable. An exponential amount of data is needed for increasing levels
+for a discrete variable.
+
+.. autosummary::
+   :toctree: generated/
+
+    GSquareCITest
+
+Kernel-Approaches
+-----------------
+Kernel-based tests are attractive since they are semi-parametric and use kernel-based ideas
+that have been shown to be robust in the machine-learning field. The Kernel CI test is a test
+that computes a test statistic from kernels of the data and uses permutation testing to
+generate samples from the null distribution :footcite:`Zhang2011`, which are then used to
+estimate a pvalue.
+
+.. autosummary::
+   :toctree: generated/
+
+    KernelCITest
+
+Classifier-based Approaches
+---------------------------
+Another suite of approaches that rely on permutation testing is the classifier-based approach.
+By shuffling data in an intelligent way, one can setup a hypothesis test for CI based on the
```

Comment:
What does this "intelligent" mean here? Wouldn't you just uniformly shuffle them in practice?

Reply:
Marginal ind testing = shuffling the data unifromly
CI testing = shuffling the data conditioned on Z (done via kNN). This is a subroutine in any permutation type procedure.

Reply:
I reworded this section tho